File: 프로그래머스/2/42586. 기능개발/기능개발.swift
import Foundation

func solution(_ progresses:[Int], _ speeds:[Int]) -> [Int] {
    var arr:[Int] = []
    for index in 0..<progresses.count {
        let remainingWork = 100 - progresses[index]
        let daysRequired = Double(remainingWork) / Double(speeds[index])
        let daysToComplete = Int(ceil(daysRequired))
        arr.append(daysToComplete)    
    }
    
    var flag = arr.removeFirst()
    var res: [Int] = [1]
    
    
    for i in arr {
        if flag >= i {
            res[res.count - 1] += 1
        } else {
            res.append(1)
            flag = i
        }
    }
    
    
    
    return res
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    progress = []
    k = 0
    
    for i,j in zip(progresses, speeds):
        progress.append(math.ceil((100-i)/j))
    
    complete = progress[0]
    cnt = 1
    
    for i in progress[1:]:
        if complete >= i:
            cnt += 1
        else:
            answer.append(cnt)
            cnt = 1
            complete = i
            
    answer.append(cnt)
    return answer